api/types/types_drivers_executor.go

```diff
@@ -132,6 +158,7 @@ type ProvidesStorageExecutorCLI interface {
 // the driver implementations of the StorageExecutor interface.
 type StorageExecutorCLI interface {
 	StorageExecutorFunctions
+	StorageExecutorWithSupported
```

Comment:
One newbie question...

Why put `StorageExecutorWithSupported` here instead of putting `StorageExecutorFunctions` in `StorageExecutorWithSupported`? This seems to modify an existing interface, such that all executors will have to be updated to be usable. Is that true? If it is true and that is your intent, read no further...

The other way around would seeming make `StorageExecutorWithSupported` a new interface that you could check if an executor implemented before calling the `Supported` method -- thereby allowing old and new versions. I may be completely off here and speaking total nonesense.